clients/pkg/logentry/stages/nonindexedlabels.go

```diff
@@ -0,0 +1,42 @@
+package stages
+
+import (
+	"github.com/go-kit/log"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/mitchellh/mapstructure"
+	"github.com/prometheus/common/model"
+)
+
+func newNonIndexedLabelsStage(params StageCreationParams) (Stage, error) {
+	cfgs := &LabelsConfig{}
+	err := mapstructure.Decode(params.config, cfgs)
+	if err != nil {
+		return nil, err
+	}
+	err = validateLabelsConfig(*cfgs)
+	if err != nil {
+		return nil, err
+	}
+	return &nonIndexedLabelsStage{
+		cfgs:   *cfgs,
+		logger: params.logger,
+	}, nil
+}
+
+type nonIndexedLabelsStage struct {
+	cfgs   LabelsConfig
+	logger log.Logger
+}
+
+func (s *nonIndexedLabelsStage) Name() string {
+	return StageTypeNonIndexedLabels
+}
+
+func (s *nonIndexedLabelsStage) Run(in chan Entry) chan Entry {
```

Comment:
The labels stage implements the Processor interface and `newLabelStage` returns a stage via the `toStage` function that takes a `Processor` and wraps it into a `Stage`. This is also done in pretty much the rest of stages.
https://github.com/grafana/loki/blob/a7094e9f19c387ab1eb64381741ce5f8d457c656/clients/pkg/logentry/stages/labels.go#L52-L55

I think we should leave this stage as close to the labels one as possible. So I think we should use the same pattern.

Moreover, the `toStage` function adds an inspector. I'm not familiar with what that is, but I wonder if that's something we want to have provided the rest of stages have it.

Reply:
Processor does not have access to the entry :
```go
type Processor interface {
	Process(labels model.LabelSet, extracted map[string]interface{}, time *time.Time, entry *string)
	Name() string
}
```
that is why I used Stage:
```
type Stage interface {
	Name() string
	Run(chan Entry) chan Entry
}
```
because non-indexed labels are stored in Entry.
JsonParser is also a stage, not a processor.